pkg/trace/config/apply.go

```diff
@@ -238,15 +238,20 @@ func (c *AgentConfig) applyDatadogConfig() error {
 		}
 	}
 
-	if config.Datadog.IsSet("bind_host") {
-		host := config.Datadog.GetString("bind_host")
-		c.StatsdHost = host
-		c.ReceiverHost = host
-	}
-	if config.Datadog.IsSet("apm_config.apm_non_local_traffic") {
-		if config.Datadog.GetBool("apm_config.apm_non_local_traffic") {
+	if config.Datadog.IsSet("bind_host") || config.Datadog.IsSet("apm_config.apm_non_local_traffic") {
+		if config.Datadog.IsSet("bind_host") {
+			host := config.Datadog.GetString("bind_host")
+			c.StatsdHost = host
+			c.ReceiverHost = host
+		}
+
+		if config.Datadog.IsSet("apm_config.apm_non_local_traffic") && config.Datadog.GetBool("apm_config.apm_non_local_traffic") {
```

Comment:
Why does the bind_host stuff need to change?

Isn't it sufficient to add an `else if config.IsContainerized() { ... }` block after the `config.Datadog.IsSet("apm_config.apm_non_local_traffic")`?

Reply:
Because if `bind_host` is set and `IsContainerized` is true, it will discard `bind_host` value, which is not what we want.

Reply:
Ok, I see. `bind_host` takes precedence.